flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/lookup/AsyncLookupJoinRunner.java

```diff
@@ -274,6 +275,11 @@ public void completeExceptionally(Throwable error) {
             realOutput.completeExceptionally(error);
         }
 
+        @Override
+        public void complete(SupplierWithException<Collection<Object>, Exception> supplier) {
+            throw new UnsupportedOperationException();
```

Comment:
same here and below? Maybe add a comment explaining why is it so?

Reply:
I added:

```
/**
   * Unsupported, because the containing classes are AsyncFunctions which don't have
   * access to the mailbox to invoke from the caller thread.
   */
```